src/platform/unix/mod.rs

```diff
@@ -592,23 +598,13 @@ impl OsIpcOneShotServer {
     pub fn new() -> Result<(OsIpcOneShotServer, String),UnixError> {
         unsafe {
             let fd = libc::socket(libc::AF_UNIX, SOCK_SEQPACKET, 0);
-            let mut path: Vec<u8>;
-            loop {
-                let path_string = CString::new(&b"/tmp/rust-ipc-socket.XXXXXX"[..]).unwrap();
-                path = path_string.as_bytes_with_nul().iter().cloned().collect();
-                if *mktemp(path.as_mut_ptr() as *mut c_char) == 0 {
-                    return Err(UnixError::last())
-                }
-
-                let (sockaddr, len) = new_sockaddr_un(path.as_ptr() as *const c_char);
-                if libc::bind(fd, &sockaddr as *const _ as *const sockaddr, len as socklen_t) == 0 {
-                    break
-                }
+            let temp_dir =  Builder::new().tempdir().unwrap();
+            let socket_path = temp_dir.path().join("socket");
+            let path_string = socket_path.to_str().to_owned().unwrap();
 
-                let errno = UnixError::last();
-                if errno.0 != libc::EINVAL {
-                    return Err(errno)
-                }
+            let (sockaddr, len) = new_sockaddr_un(path_string.as_ptr() as *const c_char);
```

Comment:
There was a misunderstanding here: you still do need to convert the string to a `CString` before passing it to `new_sockaddr_un`. My point was just that you can grab the string to be returned at the end *before* doing the conversion to `CString`, rather than converting the `CString` back to a standard `String`...